actionview/lib/action_view/helpers/asset_tag_helper.rb

```diff
@@ -364,6 +375,16 @@ def multiple_sources_tag_builder(type, sources)
           end
         end
 
+        def resolve_image_source(source, skip_pipeline)
+          if source.is_a?(Symbol) || source.is_a?(String)
```

Comment:
Yeah, `can't resolve object into URL` + the underlaying exception is good enough. If `polymorphic_url` raises that, we can remove the rescue here. Would that satisfy you, @rafaelfranca?